lib/internal/http2/core.js

```diff
@@ -1026,7 +1026,6 @@ function streamOnceReady() {
 // on the client side it is used to transmit request data.
 class Http2Stream extends Duplex {
   constructor(session, options) {
-    options.allowHalfOpen = true;
```

Comment:
I may end up pulling this particular change out... it appeared to have been causing an issue with some clients but further testing showed some other issues.